libs/notification/src/services/notification.service.ts

```diff
@@ -36,15 +69,48 @@ export class NotificationService {
             return NOT_SUPPORTED_ERROR$;
         }
 
-        return defer(() => {
-            const notification = new Notification(title, options);
-            const close$ = fromEvent(notification, 'close');
+        return from(this.createNotification(title, options)).pipe(
+            switchMap(notification => {
+                const close$ = this.fromEvent(notification, 'close');
 
-            return new Observable<Notification>(subscriber => {
-                subscriber.next(notification);
+                return new Observable<Notification>(subscriber => {
+                    subscriber.next(notification);
 
-                return () => notification.close();
-            }).pipe(takeUntil(close$));
-        });
+                    return () => notification.close();
+                }).pipe(takeUntil(close$));
+            }),
+        );
+    }
+
+    fromEvent<E extends keyof NotificationEventMap>(
+        targetNotification: Notification,
+        eventName: E,
+    ): Observable<{action: string} | void> {
+        const mapToVoid = map(() => undefined);
+
+        return this.swRegistration$.pipe(
+            switchMap(swRegistration => {
+                if (!swRegistration) {
+                    return fromEvent(targetNotification, eventName).pipe(mapToVoid);
+                }
+
+                const isTargetNotification = (notification: {timestamp?: number}) =>
```

Comment:
Can be outside `switchMap` and destructured.

Reply:
✅